profiles/common/modules/features/nexteuropa_editorial/includes/nexteuropa_editorial.tracking.inc

```diff
@@ -0,0 +1,429 @@
+  }
+}
+
+/**
+ * Gets the entity level error message for tracked changes are detected.
+ *
+ * @param array $tracking_lang
+ *   List of languages where tracked changes are detected.
+ * @param string $current_lang
+ *   The current language if it must be removed from the message.
+ *
+ * @return string
+ *   The message to display
+ *
+ * @see _nexteuropa_editorial_tracking_entity_validate()
+ */
+function _nexteuropa_editorial_tracking_entity_message($tracking_lang, $current_lang) {
+  if (isset($tracking_lang[LANGUAGE_NONE])) {
+    unset($tracking_lang[LANGUAGE_NONE]);
+    $tracking_lang[$current_lang] = $current_lang;
+  }
+  // We remove LANGUAGE_NONE because it is common to all translations.
+  unset($tracking_lang[LANGUAGE_NONE]);
+
+  $count = count($tracking_lang);
+  switch ($count) {
+    case 0:
+      // LANGUAGE_NONE case.
+      return t('The form cannot be saved because of tracked changes existing in this revision.');
+
+    case 1:
+      $implied_lang = end($tracking_lang);
+      $lang_name = locale_language_name($implied_lang);
+      $message_param = array('@lang' => $lang_name);
+      return t('The form cannot be saved because of tracked changes existing in the @lang version.',
+        $message_param);
+
+    default:
+      $language_names = array();
+      foreach ($tracking_lang as $lang) {
+        $language_names[$lang] = locale_language_name($lang);
+      }
+      $message_param = array('@lang' => implode(', ', $language_names));
+      return t('The form cannot be saved because of tracked changes existing in the @lang versions.',
+        $message_param);
+  }
+}
+
+/**
+ * Gets the widget type used by a specific field instance.
+ *
+ * @param array $field_instance
+ *   The field instance for which the related widget type.
+ *
+ * @return bool|string
+ *   Widget type; otherwise FALSE.
+ */
+function _nexteuropa_editorial_field_get_widget_type($field_instance) {
+  $widget_info = _nexteuropa_editorial_field_get_widget_info($field_instance);
+
+  if ($widget_info) {
+    return $widget_info['type'];
+  }
+
+  return FALSE;
+}
+
+/**
+ * Gets widget information for a specific field instance.
+ *
+ * @param array $field_instance
+ *   The field instance for which we need information.
+ *
+ * @return bool|array
+ *   Widget information; otherwise FALSE.
+ */
+function _nexteuropa_editorial_field_get_widget_info($field_instance) {
+  if (isset($field_instance['widget'])) {
+    return $field_instance['widget'];
+  }
+
+  return FALSE;
+}
+
+/**
+ * Checks if the saving of the entity must be blocked or not.
+ *
+ * @param object $entity
+ *   The entity to check.
+ * @param bool|FALSE $current_wbm_state
+ *   The entity current workbench moderation state (node only).
+ *
+ * @return bool
+ *   TRUE if the saving must be blocked; otherwise FALSE.
+ */
+function _nexteuropa_editorial_tracking_saving_to_block($entity, $current_wbm_state = FALSE) {
```

Comment:
Name of this function suggest that something is going to be saved to the block. Maybe:
_nexteuropa_editorial_tracking_changes_check()